### PR TITLE
refactor: Beacon remove div wrapper

### PR DESCRIPTION
**Pull request by suprko**
#### Relevant info and issue/PR links
https://app.clubhouse.io/peerio/story/16965/desktop-refactor-beacons-to-remove-div-wrapper-on-beaconed-content

#### Testing instructions
No change on user side, beacons should work the same as before.

On code side:
Currently, beacons get their size/positioning from a div wrapper on the element they are highlighting. Obviously this causes problems with, for example, CSS targeting, because we're modifying the DOM. In this PR, the `<Beacon>` component returns a clone of the child content with an added "beacon-target" class, and then we query the DOM for that class and calculate based on that element. No wrapper needed.

----
### Repository owner

- [ ] Was tested and can be merged.
- [ ] PR name follows conventional changelog format.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### CODE REVIEW

File: src/ui/shared-components/Beacon.tsx [hunk 1]
```diff
@@ -76,19 +66,22 @@ export default class Beacon extends React.Component<SpotBeaconProps | AreaBeacon
...
 
     @observable dispose: IReactionDisposer;
     @observable renderTimeout: NodeJS.Timer;
-    componentWillMount() {
+    componentDidMount() {
+        // setContentRect on mount
+        this.setContentRect();
+
         // Update `contentRect` on window resize
         window.addEventListener('resize', this.setContentRect);
```
Comment (anri-asaturov) [inline src/ui/shared-components/Beacon.tsx]: unsubscribe on unmount please

Comment (suprko) [reply src/ui/shared-components/Beacon.tsx]: ? https://github.com/PeerioTechnologies/peerio-desktop/blob/04d15bec33c2e962352ffb3ee334921b226b36e7/src/ui/shared-components/Beacon.tsx#L97

Comment (anri-asaturov) [reply src/ui/shared-components/Beacon.tsx]: hmm, couldn't find it in the diff

**Comment by anri-asaturov** [comment]
tests need fixing

**Comment by BlueMona** [comment]
Ok to merge

**Pull request merged**